registration/registration.go

```diff
@@ -168,6 +177,51 @@ func (r *Registration) CertifierInfo() *CertifierConfig {
 	return r.cfg.Certifier
 }
 
+func (r *Registration) LoadTrustedPublicKeys() error {
+	if r.cfg.Certifier == nil {
+		return ErrCertificationIsNotSupported
+	}
+
+	loadedKeys := make([][]byte, 0)
+
+	dirPath := r.cfg.Certifier.TrustedKeysDirPath
+	if dirPath == "" {
+		return ErrTrustedKeyDirPathIsNotSet
+	}
+
+	err := filepath.Walk(dirPath, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() || filepath.Ext(path) != ".key" {
+			return nil
+		}
+
+		keyData, err := os.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		key := Base64Enc{}
+		if err = key.UnmarshalFlag(string(keyData)); err != nil {
+			return err
+		}
+
+		if len(key.Bytes()) != ed25519.PublicKeySize {
+			return ErrInvalidPublicKey
+		}
+
+		loadedKeys = append(loadedKeys, key.Bytes())
+		return nil
+	})
+
+	r.trustedKeysMtx.Lock()
+	defer r.trustedKeysMtx.Unlock()
```

Comment:
I feel like this should be at the head of the method. If multiple calls to `LoadKeys` happen we only want one routine to read the disk at a time.

Reply:
I'm not sure, reading the dir in parallel is maybe inefficient from the GRPC request point of view, but it is not wrong. The mutex guards access to the keys map. On the other hand, if the critical section included reading the directory tree, it would be significantly longer which would harm efficiency in the code that we really care about - the **submits**.

Reply:
Yeah I feel like that the whole method should be mutually exclusive, not just the update of the field to avoid that issue. Wdyt?

Reply:
I mean in the case where we wait here (other routines are currently reading or writing to the `map`) we want to wait before we read from disk to avoid putting more load on the system or not? Sure this only realistically happens if someone reloads the certificates while a lot of nodes are submitting, but maybe that's not that rare.

E.g. the operator forgot to add a trusted certificate and realizes this while users are submitting and tries to fix it. In that case we want to block submissions before reading the certificates from disks or not?